### PR TITLE
[Feat] update Vault and RewardsDistributor logic with IStrategyManager

### PR DESCRIPTION
**Pull request by ludete**

### CODE REVIEW

File: contracts/interfaces/IStrategyManager.sol [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    /**
+     * @notice The total number of extant shares at a specific moment in the past.
+     */
+    function totalShares(address strategy, uint256 timepoint) external view returns (uint256);
+
+    /**
+     * @notice Distribute rewards with native token to strategyManager.
+     */
+    function rewards(uint256 amount) external payable;
```
Comment (CoderZhi) [inline contracts/interfaces/IStrategyManager.sol:56]: distributeRewards

File: contracts/interfaces/IStrategyManager.sol [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    /**
+     * @notice Distribute rewards with erc20 token to strategyManager.
+     */
+    function rewards(address token, uint256 amount) external returns(bool);
```
Comment (CoderZhi) [inline contracts/interfaces/IStrategyManager.sol:61]: distributeRewards

File: contracts/interfaces/IStrategyManager.sol [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    /**
+     * @notice The total number of extant shares
+     */
+    function totalShares(address strategy) external view returns (uint256);
+
+    /**
+     * @notice The total number of extant shares at a specific moment in the past.
+     */
+    function totalShares(address strategy, uint256 timepoint) external view returns (uint256);
```
Comment (CoderZhi) [inline contracts/interfaces/IStrategyManager.sol:51]: delete for now

**Comment by ququzone** [comment]
Plz fix checks errors

**Comment by ludete** [comment]
> Plz fix checks errors

done for ci task.

**Pull request merged**